package/MDAnalysis/coordinates/memory.py

```diff
@@ -0,0 +1,255 @@
+        if (provided_n_atoms is not None and
+            provided_n_atoms != self.n_atoms):
+                raise ValueError("The provided value for n_atoms does not match"
+                                 "the shape of the coordinate array")
+
+        self.ts = self._Timestep(self.n_atoms, **kwargs)
+        self.ts.dt = dt
+        if dimensions is not None:
+            self.ts.dimensions = dimensions
+        self.ts.frame = -1
+        self.ts.time = -1
+        self._read_next_timestep()
+
+    def set_array(self, coordinate_array, format='afc'):
+        """
+        Set underlying array in desired column format.
+
+        Parameters
+        ---------
+        coordinate_array : :class:`~numpy.ndarray object
+            The underlying array of coordinates
+        format
+            The order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        """
+        self.coordinate_array = coordinate_array
+        self.format = format
+
+    def get_array(self):
+        """
+        Return underlying array.
+        """
+        return self.coordinate_array
+
+    def _reopen(self):
+        """Reset iteration to first frame"""
+        self.ts.frame = -1
+        self.ts.time = -1
+
+    def timeseries(self, asel=None, start=0, stop=-1, step=1, format='afc'):
+        """Return a subset of coordinate data for an AtomGroup in desired
+        column format. If no selection is given, it will return a view of
+        the underlying array, while a copy is returned otherwise.
+
+        Parameters
+        ---------
+        asel : :class:`~MDAnalysis.core.AtomGroup.AtomGroup` object
+            Atom selection. Defaults to None, in which case the full set of
+            coordinate data is returned. Note that in this case, a view
+            of the underlying numpy array is returned, while a copy of the
+            data is returned whenever asel is different from None.
+        start, stop, skip : int
+            range of trajectory to access, start and stop are inclusive
+        format : str
+            the order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        """
+
+        array = self.get_array()
+        if format == self.format:
+            pass
+        elif format[0] == self.format[0]:
+            array = np.swapaxes(array, 1, 2)
+        elif format[1] == self.format[1]:
+            array = np.swapaxes(array, 0, 2)
+        elif format[2] == self.format[2]:
+            array = np.swapaxes(array, 0, 1)
+        elif self.format[1] == format[0]:
```

Comment:
for consistency, can this safely get changed to `format[0] == self.format[1]` (or whatever `self.format` gets renamed to